contracts/transfer/src/state.rs

```diff
@@ -53,136 +46,115 @@ impl TransferState {
             nullifiers: BTreeSet::new(),
             roots: ConstGenericRingBuffer::new(),
             balances: BTreeMap::new(),
-            var_crossover: None,
-            var_crossover_addr: None,
+            deposit: None,
             gas_price: None,
         }
     }
 
+    /// Mint a new phoenix note.
+    ///
+    /// This can only be called by the transfer- and stake-contracts.
+    /// If called by the `stake-contract`, this method will increase the total
+    /// amount of circulating dusk. This happens when the reward for staking
+    /// and participating in the consensus is withdrawn.
+    /// If called by the transfer-contract itself, it is important to make sure
+    /// that the minted value is subtracted from a contracts balance before
+    /// creating a phoenix-note.
     pub fn mint(&mut self, mint: Mint) -> bool {
-        // Only the stake contract can mint notes to a particular stealth
-        // address. This happens when the reward for staking and participating
-        // in the consensus is withdrawn.
-        if rusk_abi::caller() != STAKE_CONTRACT
-            && !rusk_abi::caller().is_uninitialized()
-        {
+        // why return bool?
+        let caller = rusk_abi::caller();
+        if caller != STAKE_CONTRACT && !rusk_abi::caller().is_uninitialized() {
```

Comment:
```suggestion
        if caller != STAKE_CONTRACT {
```

Achieves the same result